src/main/java/beam/analysis/plots/ModeChosenAnalysis.java

```diff
@@ -347,4 +374,40 @@ public void writeToRootCSVForReference(String fileBaseName) {
         return benchmarkData;
     }
 
+    class ModeChosenAvailableAlternatives {
+        String mode;
+        String availableModes;
+
+        public ModeChosenAvailableAlternatives(String mode, String availableModes) {
+            this.mode = mode;
+            this.availableModes = availableModes;
+        }
+
+        public String toCountString(Integer count) {
+            return mode+", "+availableModes+", "+count;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+
+            if (o == this) return true;
+            if (!(o instanceof ModeChosenAvailableAlternatives)) {
+                return false;
+            }
+
+            ModeChosenAvailableAlternatives modeChosenAvailableAlternatives = (ModeChosenAvailableAlternatives) o;
+
+            return modeChosenAvailableAlternatives.mode.equals(mode) &&
+                    modeChosenAvailableAlternatives.availableModes.equals(availableModes);
+        }
+
+        @Override
```

Comment:
what is this?

Reply:
i have to override java hashCode and equals method because this class is used as key in map